test/e2e/util.go

```diff
@@ -154,51 +160,65 @@ func conditionsMatchExpected(expected, actual map[string]string) bool {
 	return reflect.DeepEqual(expected, filtered)
 }
 
-func defaultExternalDNS(name, zoneID, zoneDomain string) operatorv1alpha1.ExternalDNS {
-	return operatorv1alpha1.ExternalDNS{
+func defaultExternalDNS(name, zoneID, zoneDomain, sourceType, routerName string) operatorv1alpha1.ExternalDNS {
+	resource := operatorv1alpha1.ExternalDNS{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: name,
 		},
 		Spec: operatorv1alpha1.ExternalDNSSpec{
 			Zones: []string{zoneID},
-			Source: operatorv1alpha1.ExternalDNSSource{
-				ExternalDNSSourceUnion: operatorv1alpha1.ExternalDNSSourceUnion{
-					Type: operatorv1alpha1.SourceTypeService,
-					Service: &operatorv1alpha1.ExternalDNSServiceSourceOptions{
-						ServiceType: []corev1.ServiceType{
-							corev1.ServiceTypeLoadBalancer,
-							corev1.ServiceTypeClusterIP,
-						},
-					},
-					AnnotationFilter: map[string]string{
-						"external-dns.mydomain.org/publish": "yes",
-					},
+		},
+	}
+
+	switch sourceType {
+	case "Service":
+		resource.Spec.Source = getServiceTypeSource(zoneDomain)
+	case "OpenShiftRoute":
+		resource.Spec.Source = getOpenshiftRouteTypeSource(routerName, zoneDomain)
+		// TODO : test with default as routername for default router
```

Comment:
This TODO kinda confirms that my question about merging both OpenShift test cases into one may need to be thought of.

Reply:
yes, we can keep only one test case, no need of two

Reply:
Yes, let's make your case work, then the other can be deleted because basically they are testing the same code path.